app/src/components/OnlineIndicator.vue

```diff
@@ -0,0 +1,39 @@
+<script setup lang="ts">
+import { Popover, PopoverButton, PopoverPanel } from "@headlessui/vue";
+import { SignalIcon, SignalSlashIcon } from "@heroicons/vue/20/solid";
+import { isConnected } from "luminary-shared";
+</script>
+
+<template>
+    <Popover class="relative">
+        <PopoverButton
+            class="flex items-center gap-1 rounded-full px-3 py-1.5 text-sm shadow-sm"
+            :class="{
+                'bg-green-100 text-green-800': isConnected,
+                'bg-yellow-100 text-yellow-800': !isConnected,
+            }"
+        >
+            <component :is="isConnected ? SignalIcon : SignalSlashIcon" class="h-4 w-4" />
+            <span class="hidden sm:block">{{ isConnected ? "Online" : "Offline" }}</span>
+        </PopoverButton>
+
+        <transition
+            enter-active-class="transition ease-out duration-100"
+            enter-from-class="transform opacity-0 scale-95"
+            enter-to-class="transform opacity-100 scale-100"
+            leave-active-class="transition ease-in duration-75"
+            leave-from-class="transform opacity-100 scale-100"
+            leave-to-class="transform opacity-0 scale-95"
+        >
+            <PopoverPanel
+                class="absolute z-10 w-screen max-w-40 -translate-x-1/2 lg:left-0 lg:max-w-72 lg:translate-x-0"
+            >
+                <div
+                    class="mt-2 rounded-md bg-white px-4 py-3 text-xs shadow-lg ring-1 ring-inset ring-zinc-900/5"
+                >
+                    {{ isConnected ? "You are online." : "You are offline." }}
+                </div>
```

Comment:
Missing dark mode here
![image](https://github.com/user-attachments/assets/04de02cb-bbd4-444e-bf57-4f392b9f97a2)

I am however wondering whether the notification should perhaps be a banner instead of an icon that uses space in the top bar.
![telegram-cloud-photo-size-4-5839471904426149424-x](https://github.com/user-attachments/assets/cfaef141-cb34-49ce-8175-d7449966d7cd)

This could perhaps be a reusable banner component that we can use to communicate different important information to the user.

@laurensgroeneveld what would you suggest?

Reply:
I think for the user the "online" status is not so interesting - usually they will be online, especially as we don't have a lot of offline content anyways (the videos can't be watched offline). In the CMS such an indicator is much more important I feel, as editors need to be aware if their content gets saved right away.

> This could perhaps be a reusable banner component that we can use to communicate different important information to the user.

This I feel would be a better solution. Just display a warning banner if the app is offline, and otherwise we don't show anything.